lib/std/zig/cross_target.zig

```diff
@@ -626,6 +626,11 @@ pub const CrossTarget = struct {
                 return .native;
             }
         }
+        // If the OS match and OS is macOS and CPU is arm64, treat always as native
+        // since we'll be running the foreign architecture tests using Rosetta2.
+        if (os_match and os_tag == .macos and builtin.cpu.arch == .aarch64) {
```

Comment:
Missing a `cpu_arch == .x86_64` ?

Reply:
No, I left it out intentionally. Rosetta is only available on arm64 macOS hosts.

Reply:
I really doubt Rosetta is able to run binaries other than those targeting x86...

Reply:
You're right. What I'm getting at is we don't officially support anything but x86_64 and aarch64 for macOS.

Reply:
I've added a preemptive check below, if you could have a look.